api/ocm/plugin/README.md

```diff
@@ -0,0 +1,151 @@
+
+The implemented standard commands have access to the plugin object and therefore
+determine whether there is an implementation for an extension point variant requested via the command execution.
+
+If there is an implementation, the CLI interface is just mapped to calls to interface functions of the particular extension point functionality.
+
+The standard implementation od the commands can be found below the package
+[`api/plugin/ppi/cmds](ppi/cmds), structured by the extension point name and its
+interface operation. (for example [upload/put](ppi/cmds/upload/put/cmd.go))
+
+## How plugins are used in the library
+
+The library includes the counterpart of the plugin-side support. It is found directly in package [`api/ocm/plugin](.). It shared the descriptor package for the plugin descriptor and provides a separate [Plugin](plugin.go) object type.
+
+Such an object provides methods for all extension point functions, which can the appropriate CLI command for its plugin executable. Again, it shields the CLI interface by providing an appropriate Go interface.
+
+To enable the usage of plugins in an [api/ocm/Context](../internal/context.go),
+the plugins must be registered at this context by calling [api/ocm/plugin/registration/RegisterExtensions](registration/registration.go).
+It used the plugin cache die attribute to setup a plugin cache by reading all the plugin descriptors of the configured plugins and providing appropriate plugin objects.
+
+For extension points requiring a static registration (like access method) it uses
+the information from the descriptors to determine the provided variants and registers appropriate proxy type implementations at the context.
+
+For extension points supporting an on-demand registration it use
+the registration handler registration feature of those extension points
+to register an appropriate registration handler using the namespace prefix `plugin`. This handler evaluate sub names to determine the plugin name and extension name in this plugin to provide an extension handler implementation proxy to for ward the handler functionality to the appropriate plugin.
+
+### Extension Proxies
+
+Regardless, whether it is a static type proxy or handler proxy, the proxy implementation keeps track of it plugin and extension name and maps
+the extension point functionality to the Go interface of the plugin representation, which the calls the appropriate CLI command of the plugin executable.
+
+The proxy implementation are found in `plugin` packages below the extension point package (for example the [access method plugin proxy](../extensions/accessmethods/plugin)).
+
+The registration handler registries are typically found in the extension point packages in (for example)) [registrations.go](../extensions/download/registration.go). The general registration handler registration handling is implemented in package [api/utils/registrations](../../utils/registrations).
```

Comment:
registration handler registries and dynamic registrations are not covered well in this document imo and need additional explanation.